test/controllers/comments_controller_test.rb

```diff
@@ -1,7 +1,45 @@
 require 'test_helper'
 
 class CommentsControllerTest < ActionController::TestCase
-  # test "the truth" do
-  #   assert true
+  test "create comment requires logged in" do 
+    post :create, :post_id => 'Tacocats'
+    assert_redirected_to new_user_session_path
+  end
+  
+  test "create comment success" do
+    u = FactoryGirl.create(:user)
+    sign_in u
+    pst = FactoryGirl.create(:post)
+    assert_difference 'pst.comments.count' do
+      post :create, :post_id => pst.id, :comment => {
+        :message => 'testing tests'
+      }
+    end
+    assert_redirected_to post_path(pst)
+    comment = Comment.last
+    assert_equal u, comment.user
+  end
+  
+  # test "create comment needs a valid post_id" do
+  #   u = FactoryGirl.create(:user)
+  #   sign_in u
+  #   post :create, :post_id => 'omg'
+  #   assert_response :not_found
   # end
```

Comment:
1) Error:
CommentsControllerTest#test_create_comment_needs_a_valid_post_id:
ActiveRecord::RecordNotFound: Couldn't find Post with id=omg
    app/controllers/comments_controller.rb:5:in `create'
    test/controllers/comments_controller_test.rb:26:in `block in <class:CommentsControllerTest>'